sympy/functions/special/singularity_functions.py

```diff
@@ -0,0 +1,197 @@
+from __future__ import print_function, division
+
+from sympy.core.function import Function, ArgumentIndexError
+from sympy.core import S, sympify, oo, diff
+from sympy.core.logic import fuzzy_not
+from sympy.core.relational import Eq
+from sympy.functions.elementary.complexes import im
+from sympy.functions.elementary.piecewise import Piecewise
+from sympy.functions.special.delta_functions import DiracDelta, Heaviside
+
+###############################################################################
+############################# SINGULARITY FUNCTION ############################
+###############################################################################
+
+
+class SingularityFunction(Function):
+    r"""
+    The Singularity functions are a class of discontinuous functions. They take a
+    variable, an offset and an exponent as arguments. These functions are
+    represented using Macaulay brackets as :
+
+    .. math::
+        SingularityFunction(x, a, n) := {\langle x - a \rangle}^n
```

Comment:
This is not easily readable at the command line. I suggest simply using:

`SingularityFunction(x, a, n) := <x - a>^n`